pkg/util/spill/agg_spill_util.go

```diff
@@ -0,0 +1,395 @@
+	var tmp [Int32Len]byte
+	*(*int32)(unsafe.Pointer(&tmp[0])) = value
+	buf = append(buf, tmp[:]...)
+	return buf
+}
+
+// SerializeUint32 serializes uint32 type
+func SerializeUint32(value uint32, buf []byte) []byte {
+	var tmp [Uint32Len]byte
+	*(*uint32)(unsafe.Pointer(&tmp[0])) = value
+	buf = append(buf, tmp[:]...)
+	return buf
+}
+
+// SerializeUint64 serializes uint64 type
+func SerializeUint64(value uint64, buf []byte) []byte {
+	var tmp [Uint64Len]byte
+	*(*uint64)(unsafe.Pointer(&tmp[0])) = value
+	buf = append(buf, tmp[:]...)
+	return buf
+}
+
+// SerializeInt64 serializes int64 type
+func SerializeInt64(value int64, buf []byte) []byte {
+	var tmp [Int64Len]byte
+	*(*int64)(unsafe.Pointer(&tmp[0])) = value
+	buf = append(buf, tmp[:]...)
+	return buf
+}
+
+// SerializeFloat32 serializes float32 type
+func SerializeFloat32(value float32, buf []byte) []byte {
+	var tmp [Float32Len]byte
+	*(*float32)(unsafe.Pointer(&tmp[0])) = value
+	buf = append(buf, tmp[:]...)
+	return buf
+}
+
+// SerializeFloat64 serializes float64 type
+func SerializeFloat64(value float64, buf []byte) []byte {
+	var tmp [Float64Len]byte
+	*(*float64)(unsafe.Pointer(&tmp[0])) = value
+	buf = append(buf, tmp[:]...)
+	return buf
+}
+
+// SerializeMyDecimal serializes MyDecimal type
+func SerializeMyDecimal(value *types.MyDecimal, buf []byte) []byte {
+	var tmp [types.MyDecimalStructSize]byte
+	*(*types.MyDecimal)(unsafe.Pointer(&tmp[0])) = *value
+	buf = append(buf, tmp[:]...)
+	return buf
+}
+
+// SerializeTime serializes Time type
+func SerializeTime(value types.Time, buf []byte) []byte {
+	var tmp [TimeLen]byte
+	*(*types.Time)(unsafe.Pointer(&tmp[0])) = value
+	buf = append(buf, tmp[:]...)
+	return buf
+}
+
+// SerializeDuration serializes Duration type
+func SerializeDuration(value gotime.Duration, buf []byte) []byte {
+	var tmp [DurationLen]byte
+	*(*gotime.Duration)(unsafe.Pointer(&tmp[0])) = value
+	buf = append(buf, tmp[:]...)
+	return buf
+}
+
+// SerializeInterface serialize interface type and return the number of bytes serialized
+func SerializeInterface(value interface{}, buf []byte) []byte {
+	switch v := value.(type) {
+	case bool:
+		buf = append(buf, BoolType)
+		return SerializeBool(v, buf)
+	case int64:
+		buf = append(buf, Int64Type)
+		return SerializeInt64(v, buf)
+	case uint64:
+		buf = append(buf, Uint64Type)
+		return SerializeUint64(v, buf)
+	case float64:
+		buf = append(buf, FloatType)
+		return SerializeFloat64(v, buf)
+	case string:
+		buf = append(buf, StringType)
+		buf = SerializeInt64(int64(len(v)), buf)
+		return append(buf, v...)
+	case types.BinaryJSON:
+		buf = append(buf, BinaryJSONType)
+		return SerializeBinaryJSON(&v, buf)
+	case types.Opaque:
+		buf = append(buf, OpaqueType)
+		buf = append(buf, v.TypeCode)
+		buf = SerializeInt64(int64(len(v.Buf)), buf)
+		return append(buf, v.Buf...)
+	case types.Time:
+		buf = append(buf, TimeType)
+		buf = SerializeUint64(uint64(v.CoreTime()), buf)
```

Comment:
> Why need to add `Type` and `Fsp`, I think these information are already included in the coretime ?

Because `Type` and `Fsp` need to be passed to `NewTime` which create a new `Time` object.

Reply:
But `coreTime` already contains information about `Type` and `Fsp`.

Reply:
> Why need to add `Type` and `Fsp`, I think these information are already included in the coretime ?

I will fix it.